library/src/main/java/bio/terra/landingzone/service/landingzone/azure/LandingZoneService.java

```diff
@@ -215,6 +215,10 @@ public AsyncJobResult<StartLandingZoneDeletion> startLandingZoneDeletionJob(
                 SamConstants.SamLandingZoneAction.DELETE),
         IS_AUTHORIZED);
 
+    // Validate landing zone actually exists - getLandingZoneRecord throws LandingZoneNotFound
```

Comment:
Wouldn't the Sam check above throw if the landing zone didn't exist?

Reply:
The deletion flight deletes the db record before it deletes the resource in Sam. So it's possible for the call to happen after the database step but before the sam step has completed.

It's okay if this throws a 403 instead of a 404 (the rawls operation will handle either). The issue is if it hits that in-between state, and gets a 202 instead.

Reply:
I'm not sure I understand, what call are you talking about that happens after the database step but before the same step?

If the Sam check above doesn't throw, then that means the landing zone resource still exists in Sam and we should be safe to try to delete. Maybe we need to update the step that deletes the database record to tolerate an already deleted record?